Add instructions on transferring assets in Kotlin/Java guide (#347)
hyperledger/iroha-2-docs: src/guide/kotlin-java.md

:::

## 6. Transferring assets

> After we have registered and minting the assets for Joe, let's transfer 

Reviewer: ```suggestion
After we have registered and minted the assets for Joe, let's transfer 
```

> some of it to another blockchain user. To do this, we will create a new 

Reviewer 2: ```suggestion
some of them to another blockchain user. To do this, we will create a new 
```

> some amount to another blockchain user. To do this, we will create a new 

Reviewer: ```suggestion
some of it to another blockchain user. To do this, we will create a new 
```

> user, register their asset in the `main` method and add asset transfer operations.

Reviewer 2: ```suggestion
user, register their asset with the `main` method and add transfer operations for the asset.
```

> user, register his asset in the `main` method and add assets transfer operations.

Reviewer: ```suggestion
user, register their asset in the `main` method and add asset transfer operations.
```

    sendTransaction.registerAccount(carl, listOf(carlKeyPair.public.toIrohaPublicKey()))
        .also { println("ACCOUNT $carl CREATED") }

    val carlAsset = "$assetDefinition$ASSET_ID_DELIMITER$carl"
    sendTransaction.registerAsset(carlAsset, AssetValue.Quantity(0))
        .also { println("ASSET $carlAsset CREATED") }

    sendTransaction.transferAsset(joeAsset, 10, carlAsset, joe.asAccountId(), joeKeyPair)
        .also { println("$joe TRANSFERRED FROM $joeAsset TO $carlAsset: 10") }
    query.getAccountAmount(joe, joeAsset).also { println("$joeAsset BALANCE: $it") }
    query.getAccountAmount(carl, carlAsset).also { println("$carlAsset BALANCE: $it") }
```

In the `sendTransaction` class, add a method for transferring assets.

```Kotlin
    suspend fun transferAsset(
        from: String,
        value: Int,
        to: String,
        admin: AccountId = this.admin,
        keyPair: KeyPair = this.keyPair
    ) {
        client.sendTransaction {
            account(admin)
            this.transferAsset(from.asAssetId(), value, to.asAssetId())
            buildSigned(keyPair)
        }.also {
            withTimeout(timeout) { it.await() }
        }
    }
```

> To check the result of the transfer of assets, add the `getAccountAmount()` method to the `Query` class

Reviewer: ```suggestion
To check the result of the asset transferring, add the `getAccountAmount()` method to the `Query` class:
```

> To check the result of the asset transferring, add the `getAccountAmount()` method to the `Query` class:

Reviewer 2: ```suggestion
To check the result of the asset transfer, add the `getAccountAmount()` method to the `Query` class:
```

```Kotlin
    suspend fun getAccountAmount(accountId: String, assetId: String): Long {
        return QueryBuilder.findAccountById(accountId.asAccountId())
            .account(admin)
            .buildSigned(keyPair)
            .let { query ->
                client.sendQuery(query).assets[assetId.asAssetId()]?.value
            }.let { value ->
                value?.cast<AssetValue.Quantity>()?.u32
            } ?: throw RuntimeException("NOT FOUND")
    }
```

> Console output should contain similar information

Reviewer: ```suggestion
Console output should contain similar information.
```
